globals/index.ts

```diff
@@ -1,11 +1,6 @@
 import { MapPlayer } from "../handles/player";
 
 export * from "./order";
-export const Players: MapPlayer[] = [];
-
-for (let i = 0; i < bj_MAX_PLAYER_SLOTS; i++) {
-  const pl = MapPlayer.fromHandle(Player(i));
-  if (pl) {
-    Players[i] = pl;
-  }
-}
+export const Players = Array.from({ length: bj_MAX_PLAYER_SLOTS }).map((_, i) =>
+  MapPlayer.fromHandle(Player(i))
+);
```

Comment:
Can we type this as `MapPlayer` instead of `MapPlayer | undefined`? Currently this change is breaking, and know that they won't be undefined.

Reply:
This now removes the 'undefined' possibility from the Players array.

Reply:
@BribeFromTheHive The code still returns a null list of players after mapping the array, which seems to be a bug in TSTL. don't need to map the array though anyways. The code I provided in my previous post works.

Reply:
@cipherxof I can add the predicate into the mapping function, but I am curious as to what you mean by 'returning a null list of players'. So no player is mapped? Or just the absent players aren't mapped? Players within the bounds of Player(0...bj_MAX_PLAYER_SLOTS - 1) should never be null, they just might be indicated as 'not playing'.

Reply:
Have you tested the code? It returns an empty list

Reply:
Wtf, I just tested it using the TS2Lua online playground, sure enough this:
```ts
export const Players = Array.from({ length: 16 }, (_, i) =>
  i
);
Players.forEach(i => print(1, i))
```
Prints the numbers, whereas this does not:
```ts
export const Players = Array.from({ length: 16 }).map((_, i) =>
  i
);
Players.forEach(i => print(1, i))
```

I'll make the changes to the PR, but have you submitted this as a bug to them? I will do it myself if not. This is what their code spits out as the translation:
```ts
    function __TS__ArrayFrom(arrayLike, mapFn, thisArg)
        local result = {}
        if mapFn == nil then
            for ____, v in arrayLikeIterator(arrayLike) do
                result[#result + 1] = v
            end
        else
            for i, v in arrayLikeIterator(arrayLike) do
                result[#result + 1] = mapFn(thisArg, v, i - 1)
            end
        end
        return result
    end
```
As I clearly see now, the array is entirely empty because Lua treats all of these elements as nothing at all, because setting a nil value just does nothing to a Lua table, whereas in JavaScript it actually does something. So this doesn't look like a fixable bug with TypeScript2Lua, and we can only work around it.